src/Telemetry/Tests/OutOfProcess/SimpleYamlSerializerTests.cs

```diff
@@ -0,0 +1,136 @@
+    stringPropPii: !String+Pii stringPropValue{seed}
+    longPropPii: !Long+Pii 123
+    doublePropPii: !Double+Pii 123.123
+    dateTimePropPii: !DateTime+Pii 8/12/2021 8:09:10 AM
+    boolPropPii: !Boolean+Pii True
+    guidPropPii: !Guid+Pii 00000000-037a-01c8-0102-030405060708
+";
+            Assert.AreEqual(expectedSerializedResults, serializedText);
+
+            var unexpectedCommandType =
+@"- command: !MyNewCommand
+    longProp: !Long 123
+    boolPropPii: !Boolean+Pii True
+";
+            deserializedResults = yamlSerializer.Read(EnumerableToAsyncEnumerable(unexpectedCommandType));
+            commandResults = (await AsyncEnumerableToEnumerable(deserializedResults)).ToList();
+            Assert.AreEqual(0, commandResults.Count);
+
+            var unexpectedLines =
+@"--- # line to ignore
+
+another line to ignore
+
+- command: !LogEvent
+    __name__: !String eventName0
+    stringProp: !String stringPropValue0
+    stringMultilineProp: !String line1_0__\r\n__line2__\r\n__line3
+
+    - one more line to ignore
+    longProp: !Long 123
+    doubleProp: !Double 123.123
+    dateTimeProp: !DateTime 8/12/2021 8:09:10 AM
+
+- command: !LogEvent
+    __name__: !String eventName1
+  yet another line to ignore
+    stringProp: stringPropValue1
+
+";
+            deserializedResults = yamlSerializer.Read(EnumerableToAsyncEnumerable(unexpectedLines));
+            var logEventCommandResults = (await AsyncEnumerableToEnumerable(deserializedResults))
+                .OfType<OutOfProcessLogEventCommand>().ToList();
+            Assert.AreEqual(2, logEventCommandResults.Count);
+            Assert.AreEqual("eventName0", logEventCommandResults[0].Args.Name);
+            Assert.AreEqual("stringPropValue0", logEventCommandResults[0].Args.Properties["stringProp"]);
+            Assert.AreEqual("eventName1", logEventCommandResults[1].Args.Name);
+            Assert.AreEqual("stringPropValue1", logEventCommandResults[1].Args.Properties["stringProp"]);
+        }
+
+        [TestMethod]
+        public async Task TestSimpleYamlSerializerMultipleCommands()
+        {
+            var yamlSerializer = new SimpleYamlSerializer();
+
+            List<OutOfProcessCommand> commands = new();
+            commands.AddRange(CreateOutOfProcessSetContextCommands());
+            commands.AddRange(CreateOutOfProcessLogEventCommands());
+            commands.Add(new OutOfProcessQuitCommand());
+
+            var serializedResults = yamlSerializer.Write(commands).ToList();
+            var deserializedResults = yamlSerializer.Read(EnumerableToAsyncEnumerable(serializedResults));
+            var commandResults = (await AsyncEnumerableToEnumerable(deserializedResults)).ToList();
+
+            Assert.AreEqual(commands.Count, commandResults.Count);
+            for (int i = 0; i < commands.Count; i++)
+            {
+                Assert.AreEqual(commands[i], commandResults[i]);
+            }
+        }
+    }
+}
```

Comment:
We should add more tests for invalid cases/usage. For example:
* when we pass multiple commands not seperated by blank lines
* when the client tries to log an event but the logmanager process is not there (for example, it crashed)
* when the client sends an invalid payload

Reply:
The TestSimpleYamlSerializerSingleCommand test method tests for invalid payloads, commands, indentation, etc.
The behavior of the parser, by design, is to ignore anything that is invalid and print the errors to the debug console without throwing exceptions.

We actually allow multiple commands to not be separated by blank lines, but I added a case for that anyway.

The serializer is just a serializer and is unaware of an external process.

The case of handling an external process crash is handled by the `OutOfProcessClient` class and will be tested in by the `OutOfProcessTests` class.
I'll add more test cases there, including with a real external process, but I want to do it in another PR.